### PR TITLE
remove npm dependencies in favour of bower dependencies

### PR DESCRIPTION
**Pull request by maxcbc**
## What
- removes npm dependencies
- adds bower dependencies
- updates `date-fns` references to resolve to bower `date-fns` dependency
- adds pre-test script to prevent addition of npm dependencies
- removes unused lodash dependency

## Why
This library is consumed as a bower dependency (so that its own bower dependencies are downloaded along side it). As a result it's npm dependencies won't be resolved properly by n-ui. This has caused deployment problems.

### CODE REVIEW

File: scripts/prevent-npm-deps.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+'use strict';
+
+const {dependencies = {}} = require('../package');
+
+if (Object.keys(dependencies).length > 0) {
+	global.console.error('NPM dependencies included in bower only js project, please remove');
```
Comment (robsquires) [inline scripts/prevent-npm-deps.js]: nice 👌

File: package.json [hunk 2]
```diff
@@ -77,19 +78,12 @@
...
     "sinon-chai": "^3.2.0",
-    "snyk": "^1.216.5",
-    "babel-preset-env": "^1.7.0",
-    "regenerator-runtime": "^0.13.3"
-  },
-  "dependencies": {
-    "date-fns": "2.7.0",
-    "form-serialize": "^0.7.2",
-    "js-cookie": "^2.2.0",
-    "lodash": "^4.17.10"
```
Comment (robsquires) [inline package.json]: guess this wasn't being used anyway?

Comment (maxcbc) [reply package.json]: well they were but inconsistently, if another dependency of a consuming project had these as dependencies then yes. Module resolution fun.

Comment (maxcbc) [reply package.json]: They're all (apart from lodash) now included as bower dependencies.

Comment (robsquires) [reply package.json]: Ideal

**Comment by thurstontye** [comment]
@maxcbc can you hold off releasing this. It makes sense to simpliy but I'm not entirely sure this is the correct stragegy. Have you discussed this with @i-like-robots and core ui? I want to be sure this aligned with the chnaged they are making around pagekit and n-ui before releasing.

**Comment by maxcbc** [comment]
@thurstontye I've already done a beta release but will hold off rolling out further

**Pull request merged**